core/nix.go

```diff
@@ -278,6 +278,25 @@ func NixInit() error {
 	if err != nil {
 		return err
 	}
+
+	if allowUnfree {
+		systemEnvDir := path.Join(homedir, ".config", "environment.d")
+
+		err = os.MkdirAll(systemEnvDir, 0755)
+		if err != nil {
+			log.Default().Printf("error creating user environment configuration directory")
```

Comment:
I think returning a `fmt.Errorf()` with this message + the error at the end would be better. 

Also please capitalize your strings >:(

Reply:
Go error messages should not be capitalized according to the go style guide.